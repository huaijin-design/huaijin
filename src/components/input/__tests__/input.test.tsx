import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Input } from '..';

describe('Input', () => {
  it('renders correctly', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> label', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        label='hello'
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> placeholder', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        placeholder='hello'
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> error', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        error
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> required', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        required
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> disabled', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        disabled
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders style rules -> password', () => {
    const json = renderer.create(
      <Input
        value=''
        onChange={() => {}}
        password
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });
});
