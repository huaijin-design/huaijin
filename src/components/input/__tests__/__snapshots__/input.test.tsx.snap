// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2:hover > label {
  color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      value=""
    >
      
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> disabled 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
  border-bottom-color: #d9d9d9;
  border-bottom-style: dotted;
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
  color: #d9d9d9;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
    disabled={true}
  >
    <label
      className="emotion-0"
      disabled={true}
      value=""
    >
      
    </label>
    <input
      className="emotion-1"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> error 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
  -webkit-transform: scale3d(1,1,1);
  -ms-transform: scale3d(1,1,1);
  transform: scale3d(1,1,1);
  border-color: #ff1010;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
  color: #ff1010;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      value=""
    >
       Error
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> label 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2:hover > label {
  color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      value=""
    >
      hello
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> password 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2:hover > label {
  color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      value=""
    >
      
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="password"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> placeholder 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2:hover > label {
  color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
  -webkit-transform: scale(0.75) translate3d(0,-18px,0);
  -ms-transform: scale(0.75) translate3d(0,-18px,0);
  transform: scale(0.75) translate3d(0,-18px,0);
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      placeholder="hello"
      value=""
    >
      
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="hello"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`Input renders style rules -> required 1`] = `
.emotion-3 {
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  position: relative;
}

.emotion-2::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  border-bottom: 2px solid rgba(64,64,64,0.4200000000000001);
}

.emotion-2:hover::before {
  border-color: rgba(64,64,64,0.72);
}

.emotion-2:hover > label {
  color: rgba(64,64,64,0.72);
}

.emotion-2::after {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  pointer-events: none;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0,0,0.2,1) 0ms;
  -webkit-transform: scale3d(0,1,1);
  -ms-transform: scale3d(0,1,1);
  transform: scale3d(0,1,1);
  border-bottom: 2px solid #ff6f61;
}

.emotion-0 {
  max-width: 90%;
  font-size: 16px;
  position: absolute;
  left: 0;
  top: 6px;
  z-index: -1;
  -webkit-transform-origin: top left;
  -ms-transform-origin: top left;
  transform-origin: top left;
  -webkit-transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  transition: 300ms cubic-bezier(0.25,0.8,0.5,1);
  color: rgba(64,64,64,0.4200000000000001);
}

.emotion-1 {
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  border: none;
  outline: none;
  padding: 8px 0 8px;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <label
      className="emotion-0"
      value=""
    >
       *
    </label>
    <input
      className="emotion-1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
</div>
`;
